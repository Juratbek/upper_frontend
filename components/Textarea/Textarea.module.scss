.textarea {
    width: 100%;
    height: 90px;
    padding: .4em;
}