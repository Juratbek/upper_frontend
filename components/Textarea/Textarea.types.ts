import { HTMLAttributes } from 'react';

export type TTextareaProps = HTMLAttributes<HTMLTextAreaElement>;
