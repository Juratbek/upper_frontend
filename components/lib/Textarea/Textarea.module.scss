.textarea {
    width: 100%;
    height: 90px;
    padding: .4em;
    resize: none;
    border-radius: var(--smallBorderRadius);
    border: 1px solid var(--lightBlack);
    background-color: transparent;
    color: inherit;
    outline: transparent solid 1px;

    &--transparent {
        border: none;
        outline: none;
    }
}