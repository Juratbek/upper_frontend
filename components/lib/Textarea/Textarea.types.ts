import { HTMLAttributes } from 'react';

type TColor = 'transparent' | 'dark';
export interface ITextareaProps extends HTMLAttributes<HTMLTextAreaElement> {
  color?: TColor;
}
