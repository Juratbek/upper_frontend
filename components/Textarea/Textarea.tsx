import { FC } from 'react';
import { getClassName } from 'utils';

import classes from './Textarea.module.scss';
import { TTextareaProps } from './Textarea.types';

export const Textarea: FC<TTextareaProps> = ({ className, ...props }) => {
  const rootClassName = getClassName(classes.textarea, className);

  return <textarea {...props} className={rootClassName} />;
};
