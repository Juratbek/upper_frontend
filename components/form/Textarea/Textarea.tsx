import { useTheme } from 'hooks';
import { forwardRef } from 'react';
import { getClassName } from 'utils';

import classes from './Textarea.module.scss';
import { ITextareaProps } from './Textarea.types';

export const Textarea = forwardRef<HTMLTextAreaElement, ITextareaProps>(function withRef(
  { className, color, ...props },
  ref,
) {
  const rootClassName = getClassName(classes.textarea, className, classes[`textarea--${color}`]);
  const { themeColors } = useTheme();

  return (
    <textarea
      style={{ borderColor: themeColors.input.border }}
      rows={3}
      {...props}
      className={rootClassName}
      ref={ref}
    />
  );
});
