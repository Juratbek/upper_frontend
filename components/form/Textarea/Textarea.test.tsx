import { render } from '@testing-library/react';

import { Textarea } from './Textarea';

describe('Textarea', () => {
  it('snapshot', () => {
    render(<Textarea />);
    expect(document.body).toMatchSnapshot();
  });
});
