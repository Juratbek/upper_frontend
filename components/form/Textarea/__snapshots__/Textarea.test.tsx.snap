// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Textarea > snapshot 1`] = `
<body>
  <div>
    <textarea
      class="textarea textarea--undefined"
      rows="3"
      style="border-color: rgba(0, 0, 0, 0.25);"
    />
  </div>
</body>
`;
